shared/src/simulation/events/patients-counted.ts

```diff
@@ -0,0 +1,22 @@
+import { PatientCount } from '../../models/radiogram';
+import { getCreate } from '../../models/utils';
+import { IsValue } from '../../utils/validators';
+import { IsPatientCount } from '../../utils/validators/is-patient-count';
+import type { SimulationEvent } from './simulation-event';
+
+export class PatientsCountedEvent implements SimulationEvent {
+    @IsValue('patientsCountedEvent')
+    readonly type = 'patientsCountedEvent';
+
+    @IsPatientCount()
+    readonly patientCount: PatientCount;
```

Comment:
Please use `ResourceDescription<PatientStatus>` instead. (You may also update the types in the `PatientCountRadiogram` where the `PatientCount` type comes from.)